packaging/os/pkgin.py

```diff
@@ -224,41 +272,95 @@ def install_packages(module, pkgin_path, packages):
 
     module.exit_json(changed=False, msg="package(s) already present")
 
-def update_package_db(module, pkgin_path):
+def update_package_db(module):
     rc, out, err = module.run_command(
-        format_pkgin_command(module, pkgin_path, "update"))
+        format_pkgin_command(module, "update"))
 
     if rc == 0:
-        return True
+        if re.search('database for.*is up-to-date\n$', out):
+            return False, "datebase is up-to-date"
+        else:
+            return True, "updated repository database"
     else:
         module.fail_json(msg="could not update package db")
 
+def do_upgrade_packages(module, full=False):
+    if full:
+        cmd = "full-upgrade"
+    else:
+        cmd = "upgrade"
+
+    rc, out, err = module.run_command(
+        format_pkgin_command(module, cmd))
+
+    if rc == 0:
+        if re.search('^nothing to do.\n$', out):
```

Comment:
we must make sure the LANG is set correctly if we screen scrape commands stdouts. 

The safe way is to add 

~~~python
module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C', LC_CTYPE='C')
~~~

somewhere in the main() function so it will be applied in every command.